daemon/cmd/status.go

```diff
@@ -570,51 +572,39 @@ func (d *Daemon) getIdentityRange() *models.IdentityRange {
 	return s
 }
 
-func (d *Daemon) startStatusCollector(cleaner *daemonCleanup) {
+func (d *Daemon) startStatusCollector(ctx context.Context, cleaner *daemonCleanup) error {
 	probes := []status.Probe{
-		{
-			Name: "check-locks",
-			Probe: func(ctx context.Context) (interface{}, error) {
-				// nothing to do any more.
-				return nil, nil
-			},
-			OnStatusUpdate: func(status status.Status) {
-				d.statusCollectMutex.Lock()
-				defer d.statusCollectMutex.Unlock()
-				// FIXME we have no field for the lock status
-			},
-		},
 		{
 			Name: "kvstore",
 			Probe: func(ctx context.Context) (interface{}, error) {
 				if option.Config.KVStore == "" {
-					return models.StatusStateDisabled, nil
+					return &models.Status{State: models.StatusStateDisabled}, nil
 				} else {
-					return kvstore.Client().Status()
+					return kvstore.Client().Status(), nil
 				}
 			},
 			OnStatusUpdate: func(status status.Status) {
-				var msg string
-				state := models.StatusStateOk
-				info, ok := status.Data.(string)
+				d.statusCollectMutex.Lock()
+				defer d.statusCollectMutex.Unlock()
 
-				switch {
-				case ok && status.Err != nil:
-					state = models.StatusStateFailure
-					msg = fmt.Sprintf("Err: %s - %s", status.Err, info)
-				case status.Err != nil:
-					state = models.StatusStateFailure
-					msg = fmt.Sprintf("Err: %s", status.Err)
-				case ok:
-					msg = info
+				if status.Err != nil {
```

Comment:
That's a super confusing part of the collector implementation, because none of the probes actually return an error, but there's still a possibility of that error being set there if the [probe timed out](https://github.com/cilium/cilium/blob/21ad13700b0f295e4997ffe4d204a9dbcd4cf309/pkg/status/status.go#L233) (the only reason would be a deadlock). I took a look at reworking the collector logic to drop the error altogether, but it ended up being more involved than expected (especially because most unit tests of that module depend on it). Eventually, I just adopted the easiest approach and handled the error there for consistency with the other probes, although it will practically never occur.